### PR TITLE
[IA-1754] Add VPC and Firewall methods to GoogleComputeService

### PR DESCRIPTION
**Pull request by rtitle**
This is to support https://broadworkbench.atlassian.net/browse/IA-1754. For a user project, Leo will do the following:

1. Delete `default-allow-rdp` and `default-allow-ssh` firewall rules in the default network
2. Add `default-broad-allow-ssh` firewall rule in the default network, restricted to Broad internal IPs
3. Create a dedicated network and subnet for Leonardo in the us-central1 region
4. Create the Leonardo tcp:443 firewall rule in the Leo subnet

These steps are taking [these guidelines](https://docs.google.com/document/d/1adV0LC2f_GIpl3A1AeoQuNiwcP59NToIt6VYT3xRCkU/edit) into account.

No version bump because I'm only adding APIs.

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [x] Bump the version in `project/Settings.scala` `createVersion()`
- [x] Update `CHANGELOG.md` for those libraries
- [x] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [x] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala [hunk 1]
```diff
@@ -19,6 +19,8 @@ private[google2] class GoogleComputeInterpreter[F[_]: Async: StructuredLogger: T
   diskClient: DiskClient,
   zoneClient: ZoneClient,
   machineTypeClient: MachineTypeClient,
+  networkClient: NetworkClient,
```
Comment (rtitle) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala:24]: Should we break out some of these Google clients to their own service/interpreter? 🤔

Comment (kyuksel) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala:24]: Yea, it seems like there's enough security logic to justify breaking it out 👍

**Review by kyuksel — APPROVED** [review]
LGTM 👍 Breaking out the security logic is a good idea, either as part of this or a separate PR.

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala [hunk 2]
```diff
@@ -135,3 +193,10 @@ final case class ZoneName(value: String) extends AnyVal
 final case class FirewallRuleName(value: String) extends AnyVal
 final case class MachineTypeName(value: String) extends AnyVal
 final case class RegionName(value: String) extends AnyVal
+final case class NetworkName(value: String) extends AnyVal
+final case class SubnetworkName(value: String) extends AnyVal
+final case class PollOperation(op: Operation, isDone: Boolean)
```
Comment (Qi77Qi) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala]: maybe
```
final case class PollOperation(op: Operation) {
def isDone: Boolean = op.getStatus == "DONE"
}
```
?

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala [hunk 3]
```diff
@@ -111,18 +139,48 @@ object GoogleComputeService {
...
+      .setCredentialsProvider(credentialsProvider)
+      .build()
 
     for {
       instanceClient <- resourceF(InstanceClient.create(instanceSettings))
       firewallClient <- resourceF(FirewallClient.create(firewallSettings))
       diskClient <- resourceF(DiskClient.create(diskSettings))
       zoneClient <- resourceF(ZoneClient.create(zoneSettings))
       machineTypeClient <- resourceF(MachineTypeClient.create(machineTypeSettings))
+      networkClient <- resourceF(NetworkClient.create(networkSettings))
```
Comment (Qi77Qi) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala:177]: none of these clients has any `shutDown` function on them?

**Review by Qi77Qi — APPROVED** [review]

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala [hunk 4]
```diff
@@ -68,6 +74,28 @@ trait GoogleComputeService[F[_]] {
...
+
+  def getSubnetwork(project: GoogleProject, region: RegionName, subnetwork: SubnetworkName)(
+    implicit ev: ApplicativeAsk[F, TraceId]
+  ): F[Option[Subnetwork]]
+
+  def createSubnetwork(project: GoogleProject, region: RegionName, subnetwork: Subnetwork)(
+    implicit ev: ApplicativeAsk[F, TraceId]
+  ): F[Operation]
+
+  def getOperation(project: GoogleProject, operation: Operation)(
```
Comment (Qi77Qi) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala]: we're giving this function an `Operation`, and returning an `Operation`, this seems a bit suspicious. Can we pass in what we actually need? or return just the `status` of the operation?

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala (hunk 1 above)
Comment (rtitle) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala:24]: On the other hand all these clients are under `com.google.cloud.compute.v1._` which is why I left them all in `GoogleComputeService`.

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala (hunk 2 above)
Comment (rtitle) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala]: Yeah.. 👍

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala (hunk 3 above)
Comment (rtitle) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala:177]: I think `resourceF` handles the `shutdown`

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala (hunk 4 above)
Comment (rtitle) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala]: yeah maybe will pass `OperationName` instead. We're currently just using the `status` but `Operation` has a lot more fields that might be useful.. I like returning the whole object.

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala (hunk 3 above)
Comment (Qi77Qi) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeService.scala:177]: ah cool... 👍

**Pull request merged**